### PR TITLE
feat(web-vitals): Polish CLS and TTFB web vitals on Event details

### PR DESCRIPTION
**Pull request by dashed**
Screen Shot 2020-10-13 at 5 53 12 PM

Screen Shot 2020-10-15 at 3 15 22 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx [hunk 1]
```diff
@@ -11,13 +11,17 @@ import space from 'app/styles/space';
...
 
 // translate known short form names into their long forms
 const LONG_MEASUREMENT_NAMES = {
   fid: 'First Input Delay',
   fp: 'First Paint',
   fcp: 'First Contentful Paint',
   lcp: 'Largest Contentful Paint',
+  cls: 'Cumulative Layout Shift',
+  ttfb: 'Time to First Byte',
+  'ttfb.requesttime': 'Request Time',
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx:24]: These long names are already defined in `performance/realUserMonitoring/constants.tsx`, can we import them from there?

File: src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx [hunk 2]
```diff
@@ -155,4 +157,12 @@ const Value = styled('span')<{failedThreshold: boolean}>`
   }};
 `;
 
+function prettyValue(record: Vital | undefined, value: number): string {
+  if (record && record.type === 'duration') {
+    return getDuration(value / 1000, 3);
+  }
+
+  return value.toFixed(3);
+}
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx:166]: I think this was brought up before but something like this has been copied pasted in several places. Can they be consolidated somehow? Maybe not in this PR but in the future.

File: src/sentry/static/sentry/app/views/performance/realUserMonitoring/types.tsx [hunk 3]
```diff
@@ -11,6 +11,7 @@ export type Vital = {
   description: string;
   failureThreshold: number;
   type: ColumnType;
+  display: boolean;
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/realUserMonitoring/types.tsx:14]: This is just here temporarily right?

Comment (dashed) [reply src/sentry/static/sentry/app/views/performance/realUserMonitoring/types.tsx:14]: yea. don't want to load them as histograms.

File: src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx]: should be fixed in https://github.com/getsentry/sentry/pull/21316/commits/311efda0cf35c60a6627aa0917154657c6645340

File: src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx:162]: We have a few similar type formatting functions in app/utils/discover/charts.tsx as well.

File: src/sentry/static/sentry/app/utils/discover/fields.tsx [hunk 4]
```diff
@@ -485,13 +485,19 @@ export enum WebVital {
...
 
 const MEASUREMENTS: Readonly<Record<WebVital, ColumnType>> = {
   [WebVital.FP]: 'duration',
   [WebVital.FCP]: 'duration',
   [WebVital.LCP]: 'duration',
   [WebVital.FID]: 'duration',
+  [WebVital.CLS]: 'number',
+  [WebVital.TTFB]: 'duration',
+  [WebVital.RequestTime]: 'duration',
 };
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/discover/fields.tsx:501]: Should this new data be added to the `WEB_VITALS_DETAILS`? Keeping the details data and these enums aligned feels like we are making our lives hard.

Comment (Zylphrex) [reply src/sentry/static/sentry/app/utils/discover/fields.tsx:501]: Currently, `WEB_VITAL_DETAILS` reads in these values via `measurementType`, but I think you're right. To simplify this, we can have all the type information in `WEB_VITAL_DETAILS` and have `measurementType` read in the type information from there instead.

Comment (dashed) [reply src/sentry/static/sentry/app/utils/discover/fields.tsx:501]: It's already there. This is used as a source of truth and added to `WEB_VITALS_DETAILS`

File: src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/realUserMonitoring.tsx]: I added `src/sentry/static/sentry/app/utils/measurements/index.tsx` to consolidate these functions that are specific to measurements/web vitals.

File: src/sentry/static/sentry/app/views/performance/realUserMonitoring/index.tsx [hunk 5]
```diff
@@ -163,15 +163,19 @@ function generateRumEventView(
     if (isAggregateField(field)) conditions.removeTag(field);
   });
 
+  const vitals = Object.keys(WEB_VITAL_DETAILS).filter(key => {
+    return !!VITAL_GROUPS.find(group => group.group.includes(key as WebVital));
+  }) as WebVital[];
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/performance/realUserMonitoring/index.tsx:168]:
```suggestion
  const vitals = VITAL_GROUPS.reduce((allVitals: WebVital[], group) => {
    return allVitals.concat(group.group);
  }, []);
```

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**